content/BindingMethods/BindingMethods.q.md

```diff
@@ -0,0 +1,1792 @@
+
+- It duplicates the named placeholders in the `bindParam()` method.
+
+
+/// type=MS, answer=[2,3,5]
+
+Which statements correctly describe the code on line 16 of `Student.php`?
+
+- It queries the SQL statement.
+
+- It calls the `fetchAll()` method of the `$pstmt` object.
+
+- It returns the array containing all of the result set rows.
+
+- It executes the argument `$pstmt` in the `fetchAll()` method.
+
+- It assigns the returned value of `fetchAll()` method to the variable `$data`.
+
+:::
+
+
+:::
+
+/// type=REPL, readonly=true, filename=[Connection.php,Student.php]
+
+```php
+<?php
+    class Connection
+    {
+        const HOST = 'localhost';
+        const DB = 'LibraryDB';
+        const PORT = '5432';
+        const USERNAME = 'codestop';
+        const PASSWORD = 'Admin01';
+    
+        public static function getConnection()
+        {
+            try {
+                return new PDO("pgsql:host=" . self::HOST . ";port=". self::PORT . ";dbname=" . self::DB . ";user=" . self::USERNAME . ";password=" . self::PASSWORD); 
+            } catch (Exception $e) {
+                throw new Exception("Unable to establish a connection."); 
+            }
+        }
+    }
+?>
+```
+
+```php
+<?php
+    require_once("./Connection.php");
+
+    $conn = Connection::getConnection();
+    $sql = "DELETE FROM students WHERE first_name = ? AND last_name = ?";
+
+    try {
+        $pstmt = $conn->prepare($sql);
+        $pstmt->bindValue(1, 'John');
+        $pstmt->bindValue(2, 'Smith');
+        if (!$pstmt->execute()) {
+            throw new Exception("Unable to delete student data in the table.");
+        }
+        echo "Successfully deleted the student data in the table.";
+    } catch (Execption $e) {
+        echo $e->getMessage();
+    }
+?>
+```
+/// type=SS, answer=[5]
+
+Execute the program. What is its output?
+
+- It produces an error.
+
+- No output is displayed.
+
+- It prints `Unable to establish a connection.`.
+
+- It prints `Successfully connected to the database.` and `Unable to delete student data in the table.`.
+
+- It prints `Successfully connected to the database.` and `Successfully deleted the student data in the table.`.
+
+
+/// type=SS, answer=[1]
+
+Which variable holds the SQL statement in `Student.php`?
+
+- `$sql`
+
+- `$conn`
+
+- `$pstmt`
+
+- `last_name`
+
+- `first_name`
+
+
+/// type=SS, answer=[1]
+
+In `Student.php`, which of the following is a positional placeholder?
+
+- `?`
+
+- `$sql`
+
+- `students`
+
+- `DELETE FROM`
+
+- `first_name = ?`
+
+
+/// type=SS, answer=[4]
+
+In the statement `$pstmt->bindValue(1, 'John');` on line 9 of `Student.php`, what is `1`?
+
+- It represents the named placeholder.
+
+- It represents the explicit data type for the parameter.
+
+- It represents the value to bind to the positional placeholder `?` in the SQL statement. 
+
+- It represents the 1-index position of the positional placeholder `?` in the SQL statement.
```

Comment:
Maybe this is the first time encountered the term `1-index position`, is it different from the term `first index` or `second index` ?

Reply:
This is `1-index position` I already check it in the Manual. Please check here https://www.php.net/manual/en/pdostatement.bindparam.php. Thanks.